src/NUnitFramework/framework/Attributes/PlatformAttribute.cs

```diff
@@ -56,12 +56,20 @@ public class PlatformAttribute : IncludeExcludeAttribute, IApplyToTest
         /// <param name="test">The test to modify</param>
         public void ApplyToTest(Test test)
         {
-            if (test.RunState != RunState.NotRunnable && 
-                test.RunState != RunState.Ignored && 
-                !platformHelper.IsPlatformSupported(this))
+            try
             {
-                test.RunState = RunState.Skipped;
-                test.Properties.Add(PropertyNames.SkipReason, platformHelper.Reason);
+                if (test.RunState != RunState.NotRunnable &&
+                    test.RunState != RunState.Ignored &&
+                    !platformHelper.IsPlatformSupported(this))
+                {
+                    test.RunState = RunState.Skipped;
+                    test.Properties.Add(PropertyNames.SkipReason, platformHelper.Reason);
+                }
+            }
+            catch (Exception ex)
+            {
+                test.RunState = RunState.NotRunnable;
+                test.Properties.Add(PropertyNames.SkipReason, ex.Message);
             }
```

Comment:
Here we are assuming that every exception gives a simple, user-focused message without a need to display the stack-trace or show inner exceptions. So we need to catch some exception type here, but not others. Maybe we need a special exception.